pkg/configurator/validating_webhook.go

```diff
@@ -0,0 +1,346 @@
+		return
+	}
+
+	requestForNamespace, admissionResp := whc.getAdmissionReqResp(admissionRequestBody)
+
+	resp, err := json.Marshal(&admissionResp)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error marshalling admission response: %s", err), http.StatusInternalServerError)
+		log.Error().Err(err).Msgf("Error marshalling admission response; Responded to admission request for configmap in namespace %s with HTTP %v", requestForNamespace, http.StatusInternalServerError)
+		return
+	}
+
+	if _, err := w.Write(resp); err != nil {
+		log.Error().Err(err).Msgf("Error writing admission response for pod in namespace %s", requestForNamespace)
+	}
+}
+
+func (whc *webhookConfig) getAdmissionReqResp(admissionRequestBody []byte) (requestForNamespace string, admissionResp v1beta1.AdmissionReview) {
+	var admissionReq v1beta1.AdmissionReview
+	if _, _, err := deserializer.Decode(admissionRequestBody, nil, &admissionReq); err != nil {
+		log.Error().Err(err).Msg("Error decoding admission request body")
+		admissionResp.Response = webhook.AdmissionError(err)
+	} else {
+		admissionResp.Response = whc.validateConfigMap(admissionReq.Request)
+	}
+
+	return admissionReq.Request.Namespace, admissionResp
+}
+
+func (whc *webhookConfig) validateConfigMap(req *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
+	if req == nil {
+		log.Error().Msg("nil admission request")
+		return webhook.AdmissionError(errNilAdmissionRequest)
+	}
+
+	// Decode the configmap from the request
+	var configMap corev1.ConfigMap
+	if _, _, err := deserializer.Decode(req.Object.Raw, nil, &configMap); err != nil {
+		log.Error().Err(err).Msgf("Error unmarshaling request to configmap in namespace %s", req.Namespace)
+		return webhook.AdmissionError(err)
+	}
+
+	log.Trace().Msgf("Validation request: (new object: %v) (old object: %v)", string(req.Object.Raw), string(req.OldObject.Raw))
+
+	//default response
+	resp := &v1beta1.AdmissionResponse{
+		Allowed: true,
+		Result:  &metav1.Status{Reason: ""},
+		UID:     req.UID,
+	}
+
+	//Allow if not a change to osm-config
+	if req.Name != constants.OSMConfigMap {
+		return resp
+	}
+
+	whc.validate(configMap, resp)
+
+	return resp
+}
+
+//validate performs all the checks to the configmap fields and rejects as necessary
+func (whc *webhookConfig) validate(configMap corev1.ConfigMap, resp *v1beta1.AdmissionResponse) *v1beta1.AdmissionResponse {
+	for field, value := range configMap.Data {
+		if !checkBoolFields(field, value, boolFieldsInConfigMap) {
+			reasonForDenial(resp, MustBeBool, field)
+		}
+		if field == "envoy_log_level" && !checkEnvoyLogLevels(field, value, ValidEnvoyLogLevels) {
+			reasonForDenial(resp, MustBeValidLogLvl, field)
+		}
+		if field == "service_cert_validity_duration" {
+			t, err := time.ParseDuration(value)
+			if err != nil {
+				reasonForDenial(resp, MustBeValidTime, field)
+			}
+			if t.Hours() > hrInAYear {
+				reasonForDenial(resp, MustBeLessThanAYear, field)
+			}
```

Comment:
Why is this constraint being added?

Reply:
@shashankram saw that k8s normally has certs last for a year so was just following that, but i can change it if we have a cert valid period or remove if it if we don't

Reply:
I suggest we remove this and introduce this check as a separate change based on discussions.